### PR TITLE
Adopt new primer border radius

### PR DESCRIPTION
**Pull request by niik**
~~🌵 This is based on #10224, please review that first 🌵~~

## Description

In primer v15 the default border radius was increased from 3px to 6px (https://github.com/primer/css/commit/2b54e867a4006ca0fffd9b01c9c1ab689c9a34f9) along with a slew of other visual changes. As GitHub.com upgraded to primer v15 not to long ago the discrepancy between Desktop and GitHub.com has become more noticeable.

This PR takes us one tiny step towards aligning (where aligning makes sense) to Primer v15 and beyond.

### Screenshots

image (1)
image (2)
image (3)
image

## Release notes

Notes: no-notes

### CODE REVIEW

**Comment by ampinsk** [comment]
Excited to start looking at this! This already feels like a nice improvement, but would love design systems to take a look (sent to them in Slack since we can't tag them here!) 

I'm curious if there's any spacing adjustment that would make sense alongside this 🤔

**Comment by rafeca** [comment]
This looks really nice!

I've noticed a small glitch in the button corners on the screenshot of the preferences dialog, we may need to adjust some more CSS:

90414371-fefb4480-e0af-11ea-982f-f9e792e616e6

File: app/styles/ui/_tab-bar.scss [hunk 1]
```diff
@@ -115,6 +115,14 @@
     border-radius: var(--border-radius);
     min-width: 150px;
 
+    &:focus {
+      // When we use the regular border radius (6px) the outline
+      // leaves a tiny gap in the corners so we drop down to an
+      // odd number which fixes the issue for us.
```
Comment (rafeca) [inline app/styles/ui/_tab-bar.scss:121]: 😆

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F13FrpeVH09Zrb2%2Fgiphy.gif&token=css" data-gitmeme-token="css"></a>

**Review by rafeca — APPROVED** [review]
Looks good to me, the new border radius and rounded avatars fit super well with the new icons, nice refresh!

**Comment by Rexogamer** [comment]
Not sure if those commits fixed this too, but I noticed another minor (and pretty similar) error in the screenshot:
B8106F69-4F7E-408B-9394-85B31DB8CDFC

**Comment by niik** [comment]
> Excited to start looking at this! This already feels like a nice improvement, but would love design systems to take a look (sent to them in Slack since we can't tag them here!)
> 
> I'm curious if there's any spacing adjustment that would make sense alongside this 🤔

I'm sure there's plenty of adjustments that would make sense but I think it's a good improvement on its own so I'm thinking we should ship merge this and ship it to beta which has the added benefit of getting more :eyes: on it. It'll be a month before we're ready to ship to production so plenty of time to solicit feedback. Perhaps we can get some of the design systems folks to jump on the beta channel even?. Happy to get feedback about this change either here or in a new issue!

**Comment by rafeca** [comment]
I've also added eace409 to get rid of the small gaps that appeared in the `<VerticalSegmentedControl/>` component when it's focused:

Screen Shot 2020-08-25 at 12 14 36

> I'm sure there's plenty of adjustments that would make sense but I think it's a good improvement on its own so I'm thinking we should ship merge this and ship it to beta which has the added benefit of getting more 👀 on it. It'll be a month before we're ready to ship to production so plenty of time to solicit feedback. 

I completely agree: the beta channel is the perfect place to iterate and do small adjustments based on feedback.

**Review by rafeca — APPROVED** [review]

**Comment by niik** [comment]
> I've also added [eace409](https://github.com/desktop/desktop/commit/eace4098aaa77b6e68bb8c33d6cc4cb4623e04d8) to get rid of the small gaps that appeared in the `<VerticalSegmentedControl/>` component when it's focused:

Perfect, thanks!

**Review by rafeca — APPROVED** [review]

**Pull request merged**